addons/mail/static/src/messaging/component/chatter_topbar/chatter_topbar.xml

```diff
@@ -4,53 +4,62 @@
+                            <i class="fa fa-clock-o"/>
+                            Schedule activity
+                        </button>
                     </t>
-                </button>
-                <div class="o-autogrow"/>
-                <t t-if="chatter.hasFollowers">
-                    <div class="o_ChatterTopbar_rightSection">
-                        <div class="o_ChatterTopbar_followers">
-                            <button class="btn btn-link o_ChatterTopbar_buttonFollow" type="button" t-att-disabled="chatter.isDisabled" t-on-click="_onClickFollow">
-                                Follow
-                            </button>
-                            <button class="btn btn-link o_ChatterTopbar_buttonFollowers" type="button" t-att-disabled="chatter.isDisabled" t-on-click="_onClickFollowers">
-                                <i class="fa fa-user"/>
-                                <span class="o_ChatterTopbar_buttonCount o_ChatterTopbar_buttonFollowersCount pl-1">0</span>
-                            </button>
+                    <button class="btn btn-link o_ChatterTopbar_buttonAttachments" type="button" t-att-disabled="chatter.isDisabled" t-on-click="_onClickAttachments">
+                        <i class="fa fa-paperclip"/>
+                        <t t-if="chatter.isDisabled or chatter.thread and chatter.thread.areAttachmentsLoaded">
+                            <span class="o_ChatterTopbar_buttonCount o_ChatterTopbar_buttonAttachmentsCount" t-esc="chatter.thread ? chatter.thread.allAttachments.length : 0"/>
+                        </t>
+                        <t t-else="">
+                            <i class="o_ChatterTopbar_buttonAttachmentsCountLoader fa fa-spinner fa-spin"/>
+                        </t>
+                    </button>
+                    <div class="o-autogrow"/>
+                    <t t-if="chatter.hasFollowers">
+                        <div class="o_ChatterTopbar_rightSection">
+                                <div class="o_ChatterTopbar_followers">
+                                    <button class="btn btn-link o_ChatterTopbar_buttonFollow" type="button" t-att-disabled="chatter.isDisabled" t-on-click="_onClickFollow">
+                                        Follow
+                                    </button>
+                                    <button class="btn btn-link o_ChatterTopbar_buttonFollowers" type="button" t-att-disabled="chatter.isDisabled" t-on-click="_onClickFollowers">
+                                        <i class="fa fa-user"/>
+                                        <span class="o_ChatterTopbar_buttonCount o_ChatterTopbar_buttonFollowersCount pl-1">0</span>
+                                    </button>
+                                </div>
                         </div>
+                    </t>
+                </div>
+                <t t-if="props.hasCloseButton">
+                    <div class="o_ChatterTopbar_buttonClose" title="Close">
```

Comment:
There is no t-on-click to give any effect to the button?

Reply:
The click event should probably do a trigger up through ChatterContainerWrapperComponent

Reply:
No it's made in `documents_controller_mixin' at line 22 :
```
(...)
events: {
        'click .o_ChatterTopbar_buttonClose': '_onClickDocumentsCloseChatter',
        (...)
(...)
```

What would you change ?

Reply:
Here "t-on-click" method doing a "trigger up" for custom event "close chatter". (find suitable name)

On "documents_controller_mixin" listen to custom event "close chatter" with the handler "_onClickDocumentsCloseChatter" (renamed to not contain "click").

It's just adding an indirection so that the component keeps the responsibility of the click.

Reply:
This would be a better first step. Note that we may then realize it is better through the store, but I can't tell yet.

Reply:
Yes, ideally we want to move most logic to the realm of entities, but this can't be done here without making more changes to documents_controller_mixin